src/execution/local.rs

```diff
@@ -52,8 +56,12 @@ use super::AppType;
 #[derive(Clone)]
 pub struct ExecutionContext {
     config: ExecutionConfig,
+    /// Underlying `SessionContext`
     session_ctx: SessionContext,
+    /// Path to the configured DDL file
     ddl_path: Option<PathBuf>,
+    /// Dedicated executor for running CPU intensive work
+    executor: Option<DedicatedExecutor>,
```

Comment:
it might make the code cleaner if you always created a DedicatedExecutor rather than only with the FlightSQL  (a possible future enhancement)

Reply:
I was going back and forth on this.  I ended up going with this approach so there was still a "default" datafusion experience.

That being said, this is meant to be an opinionated datafusion implementation so it does probably make sense to just use that.

I also had in mind doing some benchmarks under different loads - and the baseline would be without the dedicated executor.

For the time being I will live with this code smell until I figure out next steps.